corporate/lib/stripe.py

```diff
@@ -1839,7 +1944,9 @@ def get_next_plan(self, plan: CustomerPlan) -> Optional[CustomerPlan]:
                 billing_cycle_anchor=plan.end_date,
                 status=CustomerPlan.NEVER_STARTED,
             ).first()
-        return None
+        return CustomerPlan.objects.filter(
+            customer=plan.customer, fixed_price__isnull=False, status=CustomerPlan.NEVER_STARTED
+        ).last()
```

Comment:
Do we expect this plan to have `billing_cycle_anchor=plan.end_date` too? 

Probably regardless, we could have a shared `next_plan_base_query` variable, and then the `if/else` block extends the query. Might also be nice to add a comment about how we prefer the `last` record here so that if we go through a couple revisions of a next plan in the support UI, only the last is considered? (At least I assume that's the motivation).

Reply:
I'll update this after we make a decision on https://chat.zulip.org/#narrow/stream/273-kandra-billing/topic/flat-rate.20plans.20.28self-hosted.29/near/1725242

> Do we expect this plan to have billing_cycle_anchor=plan.end_date too?

No, because this added query is used to show the scheduled new_plan in /support. `billing_cycle_anchor` will have a dummy value initially. -- https://github.com/zulip/zulip/pull/28596#discussion_r1463922812

Reply:
After the introduction of `CustomerPlanOffer`, this is now:
```python
    def get_next_plan(self, plan: CustomerPlan) -> Union[CustomerPlan, CustomerPlanOffer, None]:
        customer = plan.customer
        if plan.status == CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END:
            assert plan.end_date is not None
            return CustomerPlan.objects.filter(
                customer=customer,
                billing_cycle_anchor=plan.end_date,
                status=CustomerPlan.NEVER_STARTED,
            ).first()
        elif customer.required_plan_tier is not None:
            # Currently, the only case when a next_plan is scheduled with the
            # current_plan.status NOT SET to SWITCH_PLAN_TIER_AT_PLAN_END is a
            # fixed price plan configured via /support which the customer is yet
            # to buy or schedule a purchase.
            return get_configured_fixed_price_plan_offer(
                customer=customer, plan_tier=customer.required_plan_tier
            )
        return None
```